### PR TITLE
Implement dockerize registrant

### PR DESCRIPTION
**Pull request by jxstxn1**
Dockerize will generate a `.dockerize_build` folder containing a `main.dart` which is holding an instance of dockerize data(released with 0.5.0) and also all variable data like csp hashes and `shouldEnforceCSP`.

Big benefit: User don't have a git diff after running the app due to newly generated hashes

### CODE REVIEW

File: lib/src/data/environment_base.dart [hunk 1]
```diff
@@ -0,0 +1,9 @@
+abstract class EnvironmentBase {
+  /// The name of the environment.
+  final String name = '';
+
+  /// Whether or not to report-only Content Security Policy violations.
+  final bool shouldEnforceCSP = false;
+
+  final Map<String, dynamic> versionFileEntries = const {};
```
Comment (passsy) [inline lib/src/data/environment_base.dart:8]: Don't set default values when you define an interface. They are basically unused. Only expose getters/methods

File: template/environment.template.dart [hunk 2]
```diff
@@ -1,18 +1,37 @@
...
+    'staging',
+    versionFileEntries: {
+      'environment': 'staging',
+      // You can add any other information you want to the version.json file here
+    },
+  );
   static const DockerizeEnvironment prod = DockerizeEnvironment(
     'prod',
+    versionFileEntries: {
+      'environment': 'staging',
```
Comment (Giuspepe) [inline template/environment.template.dart:32]:
```suggestion
      'environment': 'prod',
```

File: README.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ with having [docker](https://www.docker.com/) and [sidekick](https://pub.dev/pac
 
 ```bash
 <<your_cli>> sidekick plugins install dockerize_sidekick_plugin
-<<your_cli>> docker run -b
+<<your_cli>> docker run
```
Comment (robiness) [inline README.md:20]: 👍

**Review by robiness — APPROVED** [review]

File: template/server.template.dart [hunk 4]
```diff
@@ -39,7 +38,7 @@ Future<void> serverMain(List<String> args) async {
       .addMiddleware(middlewares([]))
       .addHandler(serveApp);
 
-  app.get('/<anything|.*>', handleGet);
+  app.all('/<anything|.*>', handleGet);
```
Comment (Giuspepe) [inline template/server.template.dart:41]: I'd rename `handleGet` to `handler` because it handles all http methods, not just get

**Pull request merged**